### DISCUSSION TITLE
Redis Cluster does not have support for pipeline (Job batch)

### DISCUSSION THREAD

**Discussion by 4ice** [laravel/framework#38983] [Ideas]
- Laravel Version: 8.61.0
- PHP Version: 8.0.8
- Database Driver & Version: Clustered Redis v6.0.5

### Description:
When trying to use job Batches  to dispatch a series of jobs to a redis cluster queue, an 500 exception is thrown.

```
Call to undefined method RedisCluster::pipeline() {"userId":2,"exception":"[object] (Error(code: 0): Call to undefined method RedisCluster::pipeline() at /project/vendor/laravel/framework/src/Illuminate/Redis/Connections/PhpRedisConnection.php:402
```

### Steps To Reproduce:

* Setup redis cluster for queue management according to these instructions: https://laravel.com/docs/8.x/redis#clusters
* try to dispatch jobs as a batch according to these instructions: https://laravel.com/docs/8.x/queues#job-batching

**Comment by 4ice** [comment]
For further information regarding phpredis, redis cluster and pipeline, it can be found here: https://github.com/phpredis/phpredis/blob/develop/cluster.markdown#pipelining

**Comment by 4ice** [comment]
> Have you tried configuring "predis/predis" instead of "phpredis/phpredis"?

I triend to switch to predis, but I got stuck in this issue: https://github.com/predis/predis/issues/695 and I gave up on that approach. 


My suggestion is that Laravel overrides the pipeline-function in `PhpRedisClusterConnection.php` and just execute the command instead of executing it via the pipeline. However my knowledge of the redis classes are not perfect, so I am not sure how it should be implemented.

**Comment by driesvints** [comment]
This is a feature request so I'm moving this to discussions.

**Comment by lk77** [comment]
phpredis is the way to go i think, predis is not heavily maintained, there is around 3 to 5 commits a year, so it's not a good idea to switch to predis.

**Comment by vladvildanov** [comment]
@4ice If it's still makes sense for you, I will recommend you switch to [Predis](https://github.com/predis/predis). Right now Predis supports by Redis organisation and we contribute a lot in maintaining and extending Predis capabilities.

You'll find pipelining with clusters and a lot of other features, you might be interested in
